### DISCUSSION TITLE
Video element in spoke not working with twitch stream

### DISCUSSION THREAD

**Discussion by elmau** [mozilla/hubs#3005] [General Questions & Discussion] [answered]
I made a scene with a video wall in spoke. I linked the video to a live stream from twitch (offline at that time), and then published the scene. Then when I made the room and went inside, started the stream, but the video never loaded. I restarted the live stream several times, but the video object never loaded the stream. I also loaded the stream spawning the url in the chat box to check the stream, and the video was live.

I'm doing testing for an upcoming event, and I need the video object is capable to load the stream, since there will be several rooms in a hubs-cloud instance streaming this video, and the request was to not let users to take control of the video. I'll do more testing with restream and other platforms tomorrow.

I don't know if this is an issue with other rmtp urls, but I think it's worth to take a look into it.

**Comment by johnshaughnessy** [comment, marked as answer]
@elmau Sorry for the delay on this. 

> I made the room and went inside, started the stream, but the video never loaded.

If you enter the room when the stream is offline, the videos do not "auto-refresh" when the stream becomes live. You'll have to refresh the page in order to get the live stream. 

I believe we also cache stream links for 5 minutes unless it is offline or there's an error. That means if you start your twitch stream, load the stream from hubs, restart your twitch stream, then reload hubs, it might not load correctly for 5 minutes. I'm not 100% sure about this failure case because I'm not sure if the stream link we get from yt-dl will change between you stopping and starting your twitch server.

All of the video links in the room also have "refresh" button which should let you avoid the need to refresh the whole hubs room page when troubleshooting. The refresh button is only for your local client though, so if you expect to have people in the room before the content of the stream starts, it is probably best to be streaming a blank page (or a "stream starting soon") page until you are ready to broadcast the "real" content.

Hope this answers your question!
